src/dmd/expression.d

```diff
@@ -5976,6 +5977,7 @@ extern (C++) final class UshrAssignExp : BinAssignExp
  */
 extern (C++) class CatAssignExp : BinAssignExp
 {
+    bool alreadyLowered;
```

Comment:
Why would the compiler attempt to lower an expression more than once?

Reply:
In this lowering commit:
```
*output = ce.expressionSemantic(sc);
```
on the final ConditionExpression does call expressionSemantic on `e1` and `e2`, IIRC. As we repurpose CatAssignExp when  lowering `CatAssignExp ==> __ctfe ? CatAssignExp : _d_arraycat` we have to be careful not to do the lowering on the newly generated CatAssignExp.

Hower, maybe `isCTFEExp` or something like that is a better term.